sklearn/decomposition/__init__.py

```diff
@@ -4,19 +4,19 @@
 this module can be regarded as dimensionality reduction techniques.
 """
 
-from .nmf import NMF, non_negative_factorization
-from .pca import PCA
-from .incremental_pca import IncrementalPCA
-from .kernel_pca import KernelPCA
-from .sparse_pca import SparsePCA, MiniBatchSparsePCA
-from .truncated_svd import TruncatedSVD
-from .fastica_ import FastICA, fastica
-from .dict_learning import (dict_learning, dict_learning_online, sparse_encode,
-                            DictionaryLearning, MiniBatchDictionaryLearning,
-                            SparseCoder)
-from .factor_analysis import FactorAnalysis
+from ._nmf import NMF, non_negative_factorization
+from ._pca import PCA
+from ._incremental_pca import IncrementalPCA
+from ._kernel_pca import KernelPCA
+from ._sparse_pca import SparsePCA, MiniBatchSparsePCA
+from ._truncated_svd import TruncatedSVD
+from ._fastica_ import FastICA, fastica
```

Comment:
should we remove the trailing ``_``?

Reply:
yes please

Reply:
We typically haven't for the other instances, maybe we can leave it for a post-process PR

Reply:
My intention is to make use of this chance to get rid of some bad file names :)

Reply:
Sure. All I'm saying is, there are other files that we have renamed to `_file_.py` that are already in master.